docs/root/configuration/http/http_filters/rbac_filter.rst

```diff
@@ -50,3 +50,4 @@ The RBAC filter emits the following dynamic metadata.
 
   shadow_effective_policy_id, string, The effective shadow policy ID matching the action (if any).
   shadow_engine_result, string, The engine result for the shadow rules (i.e. either `allowed` or `denied`).
+  envoy.log, string, Whether the request should be logged (i.e. either `yes` or `no`).
```

Comment:
Why is this a string and not a bool?

Reply:
This could be used to for a reason for logging and presence of the key indicates whether to log. E.g. RBAC filter would put "rbac" here.

Reply:
I think I would prefer we have a dynamic metadata key for the RBAC filter and make it a bool. This is the intended use of dynamic metadata, which avoids having to coordinate on key namespace across arbitrary number of filters.

Reply:
+1 to make it a bool.

Reply:
What about the scenarios where multiple filters can give a decision to logging filter to log/not. For example, we have a wasm sampling filter that also decides whether a request should be logged or not. It would be good to have a key that is generic enough and not filter specific so that the consuming filter can make a decision just based on the value of this key rather than needing to have knowledge of all filters from whom, it can get a decision of, to log or not.

Reply:
We can put such keys under a namespace like "envoy.common" for keys that should be read and set by multiple filters.

Reply:
I think you need to come up with a clear specification of how filters are intended to communicate here. It seems the semantics of "log" are unclear (it depends on who is the producer and who is the consumer). I'd like to see an operator docs page added that explain the new namespace and how different kinds of filters will produce and consume in it.

Reply:
There are three options to host this metadata value:
1. RBAC filter name. This is reasonable and fine for now, but it's not convenient to use once one has many filters at once. We have potentially two filters that want to make access loggers log stuff.
2. Logging filter name. This is also OK, although not common to have one filter to write into another filter's namespace. However, we have >=3 access loggers might consume this information, and it's not clear which one to pick.
3. Intermediary key, such as `envoy.common.access_log_policy`. We should document this in the metadata description, indicating that multiple filters may produce it, and multiple filters may consume it. It becomes a contract that extension developers follow. The agreement is that the metadata key should be used to force access logging from a filter, and it should be usable as once of the conditions in the access log (which deserves a mention in the operations guide).

Reply:
So, about (3), what is an access log?

This might sound like a silly question, but we have a bunch of different access logging in Envoy, including the core access logging, upstream access logging, and filter-specific (I assume you are talking here about Istio loggers that run as filters?).

Being very precise here would be helpful to avoid confusion.

I do agree in general that we can build out a rendezvous point in the key namespace, but this needs to be well specified.

Reply:
On envoy, I'm only aware of the gRPC and file access logging sinks. We can support those by either forcing all sinks to check the log key by modifying the base class, or providing an optional [access log filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/accesslog/v3/accesslog.proto#envoy-v3-api-msg-config-accesslog-v3-accesslogfilter) that can check the key. On Istio, as you said, we're using wasm filter extensions that would check the key.

Are there other potential places where we should consider to support the rbac decision on envoy?

Reply:
Can you clarify the semantics over which loggers in Istio would log when this is set? Which ones in general Envoy? There's no "wrong answer" here, it's a question of clearly defining what his field means to consumers and what the expected behaviors are. For example, would you potentially unconditionally log to the file access log locally but only do a network logging operation when the RBAC filter signals it? Or does this field control all forms of logging.

Reply:
Istio would log anytime this key is set to 'true', along with some other checks. For consumers, if they wish to use the log action they must check this key in order to make logging decisions. This field can be used to control all forms of access logging, but we can make it optional by making an access log filter that checks it, so one can specify this filter in the logger config for grpc, but not file access logging, or vice versa.

Reply:
Please specify in the docs how this interacts with Envoy's default file and gRPC access loggers. You might want to add a field that indicates what metadata controls logging to them potentially. I think you should also pick naming of the key space that indicates how this is supposed to be used when taking into account the variety of loggers; basically what I'm after is a solution that makes sense for Envoy independent of Istio.

Reply:
Sounds good. To clarify, we'll use the same key for use in all access loggers. It will be called something like "access_log_policy", which conveys its use for determining whether to output access logs. All loggers have the option of checking this key when invoked. For envoy access loggers specifically, we'd provide an [access log filter](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/accesslog/v3/accesslog.proto#envoy-v3-api-msg-config-accesslog-v3-accesslogfilter) that can be specified in the logger config that checks this key.

Reply:
OK, sounds good. Will this be a boolean or some string? What are the contents of this string if it's a policy?

Reply:
For flexibility purposes @gargnupur and @kyessenov preferred a string. Currently I just have it set to "yes" or "no", but potential use case for a string would be instead of "yes" to put the deciding filter name, for debugging or some other reason. I could go either way on this, whatever the general consensus is.

Reply:
We're going with a boolean value now.